Data/models/PlaylistFolder.swift

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+@objc(PlaylistFolder)
+final public class PlaylistFolder: NSManagedObject, CRUD {
+    @NSManaged public var uuid: String?
+    @NSManaged public var title: String?
+    @NSManaged public var order: Int32
+    @NSManaged public var dateAdded: Date?
+    @NSManaged public var playlistItems: Set<PlaylistItem>?
+    public static let savedFolderUUID = "7B6CC019-8946-4182-ACE8-42FE7B704C43"
+    
+    public class func frc(savedFolder: Bool) -> NSFetchedResultsController<PlaylistFolder> {
+        let context = DataController.viewContext
+        let fetchRequest = NSFetchRequest<PlaylistFolder>()
+        fetchRequest.entity = PlaylistFolder.entity(context)
+        fetchRequest.fetchBatchSize = 20
+        
+        let orderSort = NSSortDescriptor(key: "order", ascending: true)
+        let createdSort = NSSortDescriptor(key: "dateAdded", ascending: false)
+        fetchRequest.sortDescriptors = [orderSort, createdSort]
+        
+        if savedFolder {
+            fetchRequest.predicate = NSPredicate(format: "uuid == %@", savedFolderUUID)
+        } else {
+            fetchRequest.predicate = NSPredicate(format: "uuid != %@", savedFolderUUID)
+        }
+        
+        return NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: context,
+                                          sectionNameKeyPath: nil, cacheName: nil)
+    }
+    
+    public static func addFolder(title: String, uuid: String? = nil, completion: ((_ uuid: String) -> Void)? = nil) {
+        DataController.perform(context: .new(inMemory: false), save: false) { context in
+            var folderId: String
+            if let uuid = uuid, !uuid.isEmpty {
+                folderId = uuid
+            } else {
+                folderId = UUID().uuidString
+            }
+            
+            let playlistFolder = PlaylistFolder(context: context)
+            playlistFolder.title = title
+            playlistFolder.dateAdded = Date()
+            playlistFolder.order = -9999
+            playlistFolder.uuid = folderId
+            
+            PlaylistFolder.saveContext(context)
+            PlaylistFolder.reorderItems(context: context)
+            PlaylistFolder.saveContext(context)
+            
+            DispatchQueue.main.async {
+                completion?(folderId)
+            }
+        }
+    }
+    
+    public static func getOtherFoldersCount(context: NSManagedObjectContext? = nil) -> Int {
```

Comment:
`public class func get(with objectID: NSManagedObjectID) -> Favorite?`

Exists in the `Favorite` class because it cannot call synchronous functions with `WriteContext`. When using `WriteContext` you have to call `DataController.perform` which doesn't allow you to `return` any data. So it can't be used here as this function returns a value.